src/hotspot/share/gc/g1/g1CollectionSetCandidates.hpp

```diff
@@ -129,13 +134,17 @@ class G1CollectionCandidateList : public CHeapObj<mtGC> {
   }
 };
 
-// Iterator for G1CollectionSetCandidates.
+// Iterator for G1CollectionSetCandidates. Multiplexes across the marking/retained
+// region lists based on gc efficiency.
```

Comment:
Why does the iteration require ordering? Going through its use sites, I find they just need to perform SIMD like op on each element regardless of the order.

Reply:
API consistency. All iterators return them sorted in efficiency order, and if not doing this for that as well will give unnecessary surprises.
I can change this if you insist, but I can already see problems with that in the future.

Reply:
> I can already see problems with that in the future

Can you elaborate? In my mind, those two sources of candidates (marking vs retained) should never be placed in a single ranking.